### DISCUSSION TITLE
Is Custom Java Projects View possible?

### DISCUSSION THREAD

**Discussion by BarryDrez** [eclipse-theia/theia#12351] [Q&A]
Can anyone please tell me if there is a way to create a custom view that contains only the JAVA PROJECTS section of the Explorer/Navigator view?  Mainly, I just want to display the JRE System Library as well as the Referenced Libraries.  I would like to create a separate view on the bottom (I know how to do this) that contains these items.

**Comment by vince-fugnitto** [comment]
@BarryDrez thank you for the discussion, you can probably create a custom tree-view container and simply drag-and-drop the "java projects" tree-view item to that container. Else, given that the tree-view is contributed by a plugin you can fork it, update the location and publish a custom version of the plugin. You can also try to update where this plugin is contributed to (by extending the plugin-system) but it might be a little messy.

**Reply by BarryDrez** [reply]
@vince-fugnitto, thanks for your reply.  Actually, I think it would be better if I can keep the Explorer view where it is but filter out things like the Open editors and Workspace tree nodes.  Is this possible?

**Reply by vince-fugnitto** [reply]
@BarryDrez as an end-user if the `ViewContainerPart` supports moving location then you can drag-and-drop the view to other containers (ex: `open editors` to the `search-in-workspace` view):

![image](https://user-images.githubusercontent.com/40359487/228632920-aaee66cc-b3d3-499f-ba25-35d9532dd8ad.png)

The workspace files is not something that can be moved out of the `explorer` as it is disabled:

https://github.com/eclipse-theia/theia/blob/b33ab2cd5e9044c94533822f12d7f9d01a30e822/packages/navigator/src/browser/navigator-widget-factory.ts#L51-L57

If you do not want to do this as an end-user then you'll need to customise the behavior through the frameworks dependency injection mechanism by overriding the default implementation.
